tools/devtool

```diff
@@ -265,6 +265,20 @@ get_user_confirmation() {
     return 1
 }
 
+# Validate the user supplied version number.
+# It must be composed of 3 groups of integers separated by dot.
+#
+validate_version() {
+    declare version_regex="^([0-9]+.){2}[0-9]+$"
```

Comment:
This regex doesn't match the example provided in the comment above `cmd_release()`. The example matches the tag format (**v**0.1.0), while this check is for the number-only format (0.1.0).

Speaking of which, I think the tag should be included in this process automation, somehow. E.g. check if there is a tag matching the version provided to `devtool release`; or create the tag automagically (if it matches the right pattern and seems to be a good candidate for the next version).

Also, more `say`s, detailing the steps involved in the release process would be really nice. I.e., continuing on the tag idea, reminding the user to push the tag or something. Just a thought.

Reply:
I went ahead and added the tag creation logic, then realized it might cause problems because the tag needs to come on top of the last commit, which should be the one containing the other changes (swagger, toml etc). So we can't create the tag and change the version in one go - the modified files need to be committed, reviewed (during the review process the commit can "evolve"), CI-validated and pushed into master; only then can we come with the tag on top.

However creating the tag is in itself tedious and involves `sed`-like activities on the changelog, so I propose we add a `devtool tag-release` that would be run after `devtool release` is over and done with, reviewed and in master. What do you think?

Reply:
Ah, yes, you're right - didn't think it through. Tagging comes after committing the changes done here.